include/GFp/versioned.h

```diff
@@ -0,0 +1,400 @@
+#define GFp_nistz256_point_add VERSIONED(GFp_nistz256_point_add)
+#define GFp_nistz256_point_add_affine VERSIONED(GFp_nistz256_point_add_affine)
+#define GFp_nistz256_point_add_affinex VERSIONED(GFp_nistz256_point_add_affinex)
+#define GFp_nistz256_point_addx VERSIONED(GFp_nistz256_point_addx)
+#define GFp_nistz256_point_double VERSIONED(GFp_nistz256_point_double)
+#define GFp_nistz256_point_doublex VERSIONED(GFp_nistz256_point_doublex)
+#define GFp_nistz256_point_mul VERSIONED(GFp_nistz256_point_mul)
+#define GFp_nistz256_point_mul_base VERSIONED(GFp_nistz256_point_mul_base)
+#define GFp_nistz256_precomputed VERSIONED(GFp_nistz256_precomputed)
+#define GFp_nistz256_select_w5 VERSIONED(GFp_nistz256_select_w5)
+#define GFp_nistz256_select_w7 VERSIONED(GFp_nistz256_select_w7)
+#define GFp_nistz256_sqr_mont VERSIONED(GFp_nistz256_sqr_mont)
+#define GFp_nistz256_sub VERSIONED(GFp_nistz256_sub)
+#define GFp_nistz384_point_add VERSIONED(GFp_nistz384_point_add)
+#define GFp_nistz384_point_add_affine VERSIONED(GFp_nistz384_point_add_affine)
+#define GFp_nistz384_point_double VERSIONED(GFp_nistz384_point_double)
+#define GFp_nistz384_point_mul VERSIONED(GFp_nistz384_point_mul)
+#define GFp_p256_scalar_mul_mont VERSIONED(GFp_p256_scalar_mul_mont)
+#define GFp_p256_scalar_sqr_mont VERSIONED(GFp_p256_scalar_sqr_mont)
+#define GFp_p256_scalar_sqr_rep_mont VERSIONED(GFp_p256_scalar_sqr_rep_mont)
+#define GFp_p384_elem_add VERSIONED(GFp_p384_elem_add)
+#define GFp_p384_elem_div_by_2 VERSIONED(GFp_p384_elem_div_by_2)
+#define GFp_p384_elem_mul_mont VERSIONED(GFp_p384_elem_mul_mont)
+#define GFp_p384_elem_neg VERSIONED(GFp_p384_elem_neg)
+#define GFp_p384_elem_sqr_mont VERSIONED(GFp_p384_elem_sqr_mont)
+#define GFp_p384_elem_sub VERSIONED(GFp_p384_elem_sub)
+#define GFp_p384_scalar_inv_to_mont VERSIONED(GFp_p384_scalar_inv_to_mont)
+#define GFp_p384_scalar_mul_mont VERSIONED(GFp_p384_scalar_mul_mont)
+#define GFp_p384_scalar_sqr_rep_mont VERSIONED(GFp_p384_scalar_sqr_rep_mont)
+#define GFp_poly1305_blocks VERSIONED(GFp_poly1305_blocks)
+#define GFp_poly1305_emit VERSIONED(GFp_poly1305_emit)
+#define GFp_poly1305_init_asm VERSIONED(GFp_poly1305_init_asm)
+#define GFp_set_encrypt_key VERSIONED(GFp_set_encrypt_key)
+#define GFp_sha256_block_data_order VERSIONED(GFp_sha256_block_data_order)
+#define GFp_sha256_block_data_order_avx VERSIONED(GFp_sha256_block_data_order_avx)
+#define GFp_sha256_block_data_order_avx2 VERSIONED(GFp_sha256_block_data_order_avx2)
+#define GFp_sha256_block_data_order_shaext VERSIONED(GFp_sha256_block_data_order_shaext)
+#define GFp_sha256_block_data_order_ssse3 VERSIONED(GFp_sha256_block_data_order_ssse3)
+#define GFp_sha512_block_data_order VERSIONED(GFp_sha512_block_data_order)
+#define GFp_sha512_block_data_order_avx VERSIONED(GFp_sha512_block_data_order_avx)
+#define GFp_sha512_block_data_order_avx2 VERSIONED(GFp_sha512_block_data_order_avx2)
+#define GFp_size_t_align VERSIONED(GFp_size_t_align)
```

Comment:
In the case of `GFp_size_t_align`, et al., they are already defined using a macro, so that macro can just be modified to use `VERSIONED()` itself, something like this:

```c
#define DEFINE_METRICS(ty) \
  OPENSSL_EXPORT uint16_t GFp_VERSIONED(GFp_##ty##_align) = alignof(ty); \
  OPENSSL_EXPORT uint16_t GFp_VERSIONED(GFp_##ty##_size) = sizeof(ty);
```

Reply:
Sure, but I wanted to modify as little C/assembly code as possible. The currently solution requires zero modifications. I think it's a nice goal.

Reply:
> Sure, but I wanted to modify as little C/assembly code as possible. The current solution requires zero modifications. I think it's a nice goal.

Again, I think you're thinking about this in terms of minimizing the complexity of *this* PR. I am much more interested in minimizing the odds that we'll get a *future* PR wrong, but minimizing the amount of manual effort required for future PRs.